File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {
  private dnaToRnaMap = new Map([
    ["G", "C"],
    ["C", "G"],
    ["T", "A"],
    ["A", "U"]
  ]);

  toRna(dna: string): string {
    return Array.from(dna)
      .map(nucleotide => {
        if (!this.dnaToRnaMap.has(nucleotide)) {
          throw Error("Invalid input DNA.");
        }
        return this.dnaToRnaMap.get(nucleotide);
      })
      .join("");
  }
}

export default Transcriptor;

File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """

  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna_to_rna_map = %{?G => ?C, ?C => ?G, ?T => ?A, ?A => ?U}

    dna
    |> Enum.map(&Map.get(dna_to_rna_map, &1))
  end
end
